docs/src/index.md

```diff
@@ -183,57 +221,65 @@ Similarly every `pullback` returns an extra `∂self`, which for things without
 
 #### Pushforward / Pullback summary
 
-- **Pushforward:**
-    - returned by `frule`
-    - takes input space wiggles, gives output space wobbles
-    - 1 argument per original function argument + 1 for the function itself
-    - 1 return per original function return
-
 - **Pullback**
    - returned by `rrule`
    - takes output space wobbles, gives input space wiggles
    - 1 argument per original function return
    - 1 return per original function argument + 1 for the function itself
 
-#### Pushforward/Pullback and Total Derivative/Gradient
+- **Pushforward:**
+    - part of `frule`
+    - takes input space wiggles, gives output space wobbles
+    - 1 argument per original function argument + 1 for the function itself
+    - 1 return per original function return
+
+
+#### Pullback/Pushforward and Directional Derivative/Gradient
+
+The most trivial use of the `pushforward` from within `frule` is to calculate the directional derivative:
 
-The most trivial use of `frule` and returned `pushforward` is to calculate the [Total Derivative](https://en.wikipedia.org/wiki/Total_derivative):
+If we would like to know the the directional derivative of `f` for an input chage of `(1.5, 0.4, -1)`
 
 ```julia
-y, f_pushforward = frule(f, a, b, c)
-ẏ = f_pushforward(1, 1, 1, 1)  # for appropriate `1`-like perturbation.
+direction = (1.5, 0.4, -1) # (ȧ, ḃ, ċ)
+y, ẏ = frule(f, a, b, c, Zero(), 1, 0, 0)
 ```
 
-Then we have that `ẏ` is the _total derivative_ of `f` at `(a, b, c)`, written mathematically as ``df_{(a,b,c)}``
+On the basis directions one gets the partial derivatives of `y`:
+```julia
+y, dy_da = frule(f, a, b, c, Zero(), 1, 0, 0)
+y, dy_db = frule(f, a, b, c, Zero(), 0, 1, 0)
+y, dy_dc = frule(f, a, b, c, Zero(), 0, 0, 1)
+```
 
 Similarly, the most trivial use of `rrule` and returned `pullback` is to calculate the [Gradient](https://en.wikipedia.org/wiki/Gradient):
 
 ```julia
 y, f_pullback = rrule(f, a, b, c)
 ∇f = f_pullback(1)  # for appropriate `1`-like seed.
-s̄elf, ā, b̄, c̄ = ∇f
+s̄elf, ā, b̄, c̄ = ∇f
 ```
 Then we have that `∇f` is the _gradient_ of `f` at `(a, b, c)`.
 And we thus have the partial derivatives ``\overline{\mathrm{self}}, = \dfrac{∂f}{∂\mathrm{self}}``, ``\overline{a} = \dfrac{∂f}{∂a}``, ``\overline{b} = \dfrac{∂f}{∂b}``, ``\overline{c} = \dfrac{∂f}{∂c}``, including the and the self-partial derivative, ``\overline{\mathrm{self}}``.
 
 ### Differentials
```

Comment:
I think this whole section needs to be expanded on to make various points:

- `+` is the only operation guaranteed to be defined on differentials and, moreover, it's only guaranteed to be defined between differentials that are valid for the same primal type.
- `*` is only guaranteed to be defined between scalars and differentials. You can't generally multiple a differential by another differential
- `+` between primals and differentials isn't always defined, but it is for a lot of interesting cases e.g. `Real`, `Matrix{<:Real}` etc
- stuff that I've missed that we've figured out since this section was last written.

Reply:
Yep. But out of scope for this PR